pkg/cluster/fetch.go

```diff
@@ -74,7 +75,46 @@ func BuildSpecFromConfig(ctx context.Context, client Client, config *Config) (*S
 		return nil, err
 	}
 
-	versionsBundle, err := GetVersionsBundle(config.Cluster, bundles)
+	eksdReleases, err := fetchAllEksdReleases(ctx, client, config.Cluster, bundles)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewSpec(config, bundles, eksdReleases, eksaRelease)
+}
+
+func fetchAllEksdReleases(ctx context.Context, client Client, cluster *v1alpha1.Cluster, bundles *v1alpha1release.Bundles) ([]eksdv1alpha1.Release, error) {
```

Comment:
I think this whole method can be significatly simplified by using `cluster.KubernetesVersions`
same with the similar method in fetch.go